module_utils/network_lsr/argument_validator.py

```diff
@@ -2568,65 +2568,55 @@ def _ipv6_is_not_configured(connection):
-            line = line.lstrip()
+        # This parses the /etc/iproute2/rt_tables file and constructs
+        # the mapping from table aliases the table numeric IDs.
+        #
+        # It is thus similar to rtnl_rttable_a2n(), from here:
+        # https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/tree/lib/rt_names.c?id=11e41a635cfab54e8e02fbff2a03715467e77ae9#n447
+        rr = re.compile(
+            b"^\\s*(0x[0-9a-fA-F]+|[0-9]+)\\s+([a-zA-Z0-9_.-]+)(\\s*|\\s+#.*)$"
+        )
+        for line in file_content.split(b"\n"):
 
-            # skip empty lines or comments
-            if not line:
+            m = rr.match(line)
+            if not m:
                 continue
 
-            # In Python 2.x, there is no new type called `bytes`. And Python 2.x
-            # `bytes` is just an alias to the str type
-            if Util.PY3:
-                if line[0] == ord(b"#"):
-                    continue
-            else:
-                if line[0] == "#":
-                    continue
-
-            # iproute2 splits at the first space.
-            ll = line.split(b" ", 1)
-            if len(ll) != 2:
-                continue
-            line1 = ll[0]
-            line2 = ll[1]
-
-            comment_space = line2.find(b" ")
-            if comment_space >= 0:
-                # when commenting the route table entry in the same line,
-                # iproute2 only accepts one ' ', followed by '#'
-                if not re.match(b"^[a-zA-Z0-9_.-]+ #", line2):
-                    continue
-                line2 = line2[:comment_space]
-
-            # convert to UTF-8 and only accept benign characters.
-            try:
-                line2 = line2.decode("utf-8")
-            except Exception:
-                continue
+            table = m.group(1)
+            name = m.group(2)
 
-            if not cls.ROUTE_TABLE_ALIAS_RE.match(line2):
-                continue
+            name = name.decode("utf-8")
+            assert cls.ROUTE_TABLE_ALIAS_RE.match(name)
 
             tableid = None
             try:
-                tableid = int(line1)
+                tableid = int(table)
             except Exception:
-                if line.startswith(b"0x"):
+                if table.startswith(b"0x"):
                     try:
-                        tableid = int(line1[2:], 16)
+                        tableid = int(table[2:], 16)
                     except Exception:
                         pass
             if tableid is None or tableid < 0 or tableid > 0xFFFFFFFF:
                 continue
 
-            mapping[line2] = tableid
+            # In case of duplicates, the latter wins. That is unlike iproute2's
+            # rtnl_rttable_a2n(), which does a linear search over the
+            # hash table (thus, the first found name depends on the content
+            # of the hash table and the result in face of duplicates is
+            # not well defined).
```

Comment:
IMHO it would be better to fail and notify the user so that they can fix this.

Reply:
we already fail if we cannot resolve a name to a number. We only won't indicate whether the reason is due to syntax error in the rt_tables file.

But sure, follow-up patch welcome. This patch does not change any of this.

Reply:
it's a bugfix, so there is a visible change in behavior. It accepts now *more* than before, which tends to be less of a problem.

What does not change, is that invalid lines are ignored (only what makes a line invalid was relaxed)... to do anything about invalid lines is not part of this change.

Reply:
The unit test that I linked changes which table entry is parsed from the first one to the last one:

```
-        self.assertEqual(parse(b"5 x #df\n0x4 y\n7\ty"), {"x": 5, "y": 4})
+      self.assertEqual(parse(b"5 x #df\n0x4 y\n7\ty"), {"x": 5, "y": 7})
```

which is the change that is described in this comment.

Reply:
I am not against a followup patch to somehow handle errors differently. But I don't think that should be part of this patch. This patch doesn't change how invalid lines are handled (they are still ignored). It only changes, that it now considers strictly more lines as valid than before.